### ISSUE TITLE
If forking test case terminates abnormally, hypothesis will not print falsifying example

### ISSUE THREAD

**Issue by Vlad-Shcherbina** [DRMacIver/hypothesis#100]
```python
import os

import hypothesis
from hypothesis import strategies
from hypothesis.testrunners.forking import ForkingTestCase


class TestForking(ForkingTestCase):
    @hypothesis.given(strategies.integers())
    def test_abnormal(self, x):
        os._exit(1)


if __name__ == '__main__':
    TestForking().test_abnormal()
```

outputs

```
Traceback (most recent call last):
  File ".../hypothesis_forking_bug.py", line 15, in <module>
    TestForking().test_abnormal()
  File ".../.hypothesis/eval_source/hypothesis_temporary_module_....py", line 5, in test_abnormal
    return f(self, x)
  File ".../site-packages/hypothesis/core.py", line 532, in wrapped_test
    print_example=True
  File ".../site-packages/hypothesis/testrunners/forking.py", line 105, in execute_example
    raise AbnormalExit()
hypothesis.errors.AbnormalExit
```

There is no
```
Falsifying example: test_abnormal(self=<__main__.TestForking testMethod=runTest>, x=0)
```
as there would be for regular test failures.

I observed this behavior on ubuntu, cpython 3.4.3, hypothesis 1.8.0.

Could it be that composition `test_runner(reify_and_execute(..., test, ...))` in `core.py` is done in the wrong order?

**Comment by DRMacIver** [comment]
Right. The problem here is that the example is passed back to the parent process to handle (it's to do with capturing output), but this wasn't being immediately flushed so if the child died too abruptly the data was lost. https://github.com/DRMacIver/hypothesis/commit/9f9eedbd16e243aa9880a6fa669515d58af52603 fixes it, but I probably won't have time to put out a patch release as I'm about to head off to the airport. :-) Feel free to give the 1.8.x branch a try and see if it fixes your problem (it should). It's just 1.8.1 + this commit right now.

**Comment by DRMacIver** [comment]
ok. I've now released 1.8.2 which includes this fix.

**Comment by Vlad-Shcherbina** [comment]
FWIW, I confirm that your test consistently exposes the problem on my system, and flush fixes it.

Thank you!
